File: 0001-two-sum/0001-two-sum.php
class Solution {

    /**
     * @param Integer[] $nums
     * @param Integer $target
     * @return Integer[]
     */
    function twoSum($nums, $target) {
        $res = [];
        $setSum = [];
        
        for($i = 0; $i < sizeof($nums); $i++)
        {
            $diff = $target - $nums[$i];
            if(array_search($diff, $setSum) !== false)
            {
                $key = array_search($diff, $setSum);
                $res[] = $i;
                $res[] = $key;
            }
            $setSum[] = $nums[$i];
        }
        return $res;
    }
}

File: 0001-two-sum/0001-two-sum.swift
# @param {Integer[]} nums
# @param {Integer} target
# @return {Integer[]}
def two_sum(nums, target)
  dict = {}
  nums.each_with_index do |n, i|
    if dict[target - n]
      return dict[target - n], i
    end
    dict[n] = i
  end
end